### PR TITLE
OHRM5X-1067: Time sheet related bug fixes added

### PR DESCRIPTION
**Pull request by VageeshOrangeHRM**
This PR Contains following fixes:
1. OHRM5X-1067 - [Time - Employee timesheets] Default pending action list is arranged based on the alphabetical order of Employee Lastname
2. OHRM5X-1071 - [Time - Employee timesheets] In edit timesheet, when timesheet id parameter is changed to a non existing work shift number in URL a 500 error is getting displayed.

### CODE REVIEW

File: symfony/plugins/orangehrmTimePlugin/Dto/EmployeeTimesheetListSearchFilterParams.php [hunk 1]
```diff
@@ -35,9 +36,13 @@ class EmployeeTimesheetListSearchFilterParams extends FilterParams
...
     protected ?array $actionableStatesList = null;
 
+    /**
+     * @throws SearchParamException
+     */
     public function __construct()
     {
-        $this->setSortField('employee.lastName');
+        $this->setSortField('timesheet.startDate');
+        $this->setSortOrder('DESC');
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Dto/EmployeeTimesheetListSearchFilterParams.php:45]:
```suggestion
        $this->setSortOrder(ListSorter::DESCENDING);
```

File: symfony/plugins/orangehrmTimePlugin/Dto/EmployeeTimesheetListSearchFilterParams.php [hunk 2]
```diff
@@ -20,10 +20,11 @@
 namespace OrangeHRM\Time\Dto;
 
 use OrangeHRM\Core\Dto\FilterParams;
+use OrangeHRM\Core\Exception\SearchParamException;
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Dto/EmployeeTimesheetListSearchFilterParams.php:23]:
```suggestion
use OrangeHRM\Core\Exception\SearchParamException;
use OrangeHRM\ORM\ListSorter;
```

File: symfony/plugins/orangehrmTimePlugin/Controller/EditTimesheetController.php [hunk 3]
```diff
@@ -67,10 +69,14 @@ public function isCapable(Request $request): bool
                 ->getTimesheetDao()
                 ->getTimesheetById($request->attributes->getInt('id'));
             if ($timesheet instanceof Timesheet) {
+                if ($this->getUserRoleManagerHelper()->isSelfByEmpNumber($timesheet->getEmployee()->getEmpNumber())
+                    && $timesheet->getState() === 'APPROVED') {
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Controller/EditTimesheetController.php:73]: Shall we remove `$timesheet->getState() === 'APPROVED'` for the moment. Will address it under `OHRM5X-1056`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmTimePlugin/Dto/EmployeeTimesheetListSearchFilterParams.php (hunk 1 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmTimePlugin/Dto/EmployeeTimesheetListSearchFilterParams.php]: done

File: symfony/plugins/orangehrmTimePlugin/Controller/EditTimesheetController.php (hunk 3 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmTimePlugin/Controller/EditTimesheetController.php]: done

File: symfony/plugins/orangehrmTimePlugin/Dto/EmployeeTimesheetListSearchFilterParams.php (hunk 2 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmTimePlugin/Dto/EmployeeTimesheetListSearchFilterParams.php:23]: done

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**